lib/rspec/core/configuration.rb

```diff
@@ -155,7 +155,7 @@ def deprecation_stream=(value)
             "it to take effect, or use the `--deprecation-out` CLI option. " \
             "(Called from #{CallerFilter.first_non_rspec_line})"
```

Comment:
This warning seems like it's a different way to deal with the same issue that we're addressing for `output_stream`: what to do when users change `config.deprecation_stream` after the reporter has been initialized.  Could we apply the `OutputWrapper` solution to this config setting as well, and remove the warning?  Hopefully that's not much work...